### ISSUE TITLE
[Bug] 旋转无法回落竖屏导致歌词不可见：加速度计周期性重新附着清空判定状态，回落稳定窗口永远攒不满

### ISSUE THREAD

**Issue by aodianjun** [aodianjun/com.aodianjun.hyperglow.cnplus#31]
## 问题

0.3.92 修复 issue #30 后,旋转仍**无法回落竖屏**:画布停留在横屏逻辑帧,被放进竖屏视口 → **横屏/转回竖屏都看不到歌词**。

## 现场(0.3.92 / versionCode 119)

**系统为竖屏:**

```
mDisplayRotation=ROTATION_0   mBounds=Rect(0,0-1080,2400)   port
```

**HyperGlow 侧只有"切横屏",从无"回竖屏":**

```
[2026-09-17T17:23:04.303] rotation step PORTRAIT->LANDSCAPE enabled=true
[2026-09-17T17:24:11.209] rotation step LANDSCAPE->REVERSE_LANDSCAPE enabled=true
[2026-09-17T17:24:21.802] rotation step PORTRAIT->LANDSCAPE enabled=true
[2026-09-17T17:39:37.586] rotation step PORTRAIT->LANDSCAPE enabled=true
[2026-09-17T17:40:55.766] rotation step PORTRAIT->LANDSCAPE enabled=true
[2026-09-17T17:41:12.092] rotation step PORTRAIT->LANDSCAPE enabled=true
```

**整份日志(16:42-17:45)没有一条 `LANDSCAPE->PORTRAIT`。**

**布局仍按竖屏整高计算,而画布内部是横屏逻辑帧:**

```
[AodSurfaceController] Lyric surface placed rect=65..2367 stock=493..1574 zone=STOCK
[AodSurfaceController] Snapshot revision=280 visible=true lineLevel=true keepAlive=true render=true
    surface=true/0 effAlpha=1.0 alphaChain=LinearLayout:0:1.0/1.0>OverlayViewGroup:0:1.0/1.0 root=1080x2400 failed=false
```

即渲染层认为"在正常渲染、alpha=1",但内容按横屏(长边为宽)排版后被裁出竖屏可视区 → 视觉上无歌词。

## 根因

**加速度计被周期性"重新附着",而重新附着会把监视器状态清空,使回落所需的稳定样本永远攒不满。**

### 1. 反复重新附着(实测 16:42-17:45 共 10 次)

```
[2026-09-17T16:43:46.932] Accelerometer attached mode=auto settleMs=1000
[2026-09-17T16:44:09.297] Accelerometer attached mode=auto settleMs=1000
[2026-09-17T16:44:26.735] Accelerometer attached mode=auto settleMs=1000
[2026-09-17T16:44:58.394] Accelerometer attached mode=auto settleMs=1000
[2026-09-17T17:22:11.861] Accelerometer attached mode=auto settleMs=1000
[2026-09-17T17:24:18.574] Accelerometer attached mode=auto settleMs=1000
[2026-09-17T17:25:41.355] Accelerometer attached mode=auto settleMs=1000
[2026-09-17T17:39:35.309] Accelerometer attached mode=auto settleMs=1000
[2026-09-17T17:40:53.892] Accelerometer attached mode=auto settleMs=1000
[2026-09-17T17:41:06.028] Accelerometer attached mode=auto settleMs=1000
```

每次附着后 2-6 秒内必出现一条 `PORTRAIT->LANDSCAPE`,与"清空后重新判定"的时序完全吻合。

### 2. `attach()` 走首次附着路径时会清空判定状态

```kotlin
fun attach(context, newMode, newSettleMs, rotationListener) {
    mode = newMode; settleMs = newSettleMs; listener = rotationListener
    if (attached) {
        evaluate(resolveAodRotationStep(mode, filteredX, filteredY))   // 已附着:仅 rebase
        return
    }
    ...
    filteredX = 0f
    filteredY = 0f
    currentStep = null          // ← 清空当前朝向
    candidateStep = null
    candidateSinceElapsedMs = 0L
    stableSamples = 0
    attached = manager.registerListener(...)
    HookLogger.i(TAG, "Accelerometer attached mode=$newMode settleMs=$newSettleMs")   // ← 每次重新附着都会打
}
```

### 3. 回落需要连续稳定样本,而状态被周期性清零

```kotlin
private fun evaluate(target: AodOrientationStep?) {
    if (target == null) { stableSamples = 0; return }
    if (target != currentStep) {
        if (target != candidateStep) {
            candidateStep = target
            candidateSinceElapsedMs = SystemClock.elapsedRealtime()
            stableSamples = 0                      // ← 候选变化即清零
        } else {
            stableSamples = ((SystemClock.elapsedRealtime() - candidateSinceElapsedMs) / STABLE_SAMPLE_MS).toInt()
        }
        val required = ((settleMs / STABLE_SAMPLE_MS).toInt()).coerceAtLeast(1)   // settleMs=1000, STABLE=75 → 13
        if (shouldEmitRotationStep(currentStep, candidateStep, stableSamples, required)) {
            currentStep = target; candidateStep = null; stableSamples = 0
            listener?.invoke(target)
        }
    } else { candidateStep = null; stableSamples = 0 }
    rearmFlush()
}
```

回落 `LANDSCAPE → PORTRAIT` 需要连续 13 个样本(约 1 秒)。而 `attach` 的重新附着把 `currentStep` 置 null、并把 `filteredX/Y` 归零,候选窗口随之重置——只要重新附着周期短于/接近该窗口,回落就**永远无法落地**。

### 4. 重新附着的来源:`!rotate` 时 detach

`AodSurfaceController.applyRotation`(每次快照更新都会执行):

```kotlin
if (rotate && !AodOrientationMonitor.INSTANCE.isAttached()) {
    AodOrientationMonitor.INSTANCE.attach(context, mode, settle) { step -> mainHandler.post { onOrientationStepResolved(step) } }
} else if (!rotate) {
    AodOrientationMonitor.INSTANCE.detach()                        // ← 关闭时卸载
    onOrientationStepResolved(AodOrientationStep.PORTRAIT)
} else {
    AodOrientationMonitor.INSTANCE.attach(context, mode, settle) { ... }   // 已附着:rebase
}
```

即 **`snapshot.aodRotateWithDevice` 在 true/false 间抖动**(快照可见性变化、暂停保留窗口等都会重发快照)时,`attach`/`detach` 反复切换;每次从 detach 回到 attach 都走"首次附着"分支,清空判定状态。

### 5. 结果

- 重新附着后 `currentStep = null`,若此刻设备恰好侧放 → 立刻 emit `LANDSCAPE`
- `AodSurfaceController.onOrientationStepResolved`:
  ```kotlin
  if (currentRotationStep == step) return      // 已是 LANDSCAPE → 直接 return,画布不再更新
  ```
  于是画布停在横屏
- 转回竖屏时,回落窗口被下一次重新附着打断 → 不 emit → **画布一直横屏 → 歌词被裁不可见**

## 影响

- 侧放后回正,歌词不恢复;竖屏下画布仍是横屏逻辑帧
- 尤其影响 AOD 息屏歌词的可用性(横屏/竖屏都看不到)

## 建议

1. **重新附着时保留判定状态**:首次附着路径不要无条件 `currentStep = null`;若 `mode`/`settleMs` 未变且已有有效朝向,应沿用并只做 rebase
2. **`filteredX/Y` 归零改为"标记未初始化"**:用一个 `filteredInitialized` 标志,首个采样直接赋值而非从 0 慢收敛;避免归零瞬间被判成 PORTRAIT(0,0) 造成无谓跳变
3. **回落不应被 rebase 打断**:`attach` 的已附着分支已调 `evaluate`,但候选窗口在重新附着时被清;建议把 `candidateStep/candidateSinceElapsedMs/stableSamples` 与"附着生命周期"解耦
4. **`onOrientationStepResolved` 增加幂等补偿**:`currentRotationStep == step` 时也确保画布 `rotationStep` 与之一致(画布可能因重建而回到 PORTRAIT 默认值)
5. **排查 `aodRotateWithDevice` 抖动**:确认快照里该字段为何在短时间内反复变化(暂停/保留窗口/可见性切换时应当保持稳定)

## 备注

- 相关:issue #30(横屏无法回落竖屏)、#29(模式未生效)、#28(横屏布局未重排)
- 0.3.92 已把 `resolveAodRotationStep` 的竖持分支改为显式返回 `PORTRAIT`(代码正确),本条是**附着生命周期**层面的打断,与前几条不重复
- 若需要,我可以补充 `attach`/`detach` 调用点的完整时间线日志
